ProcessLib/BoundaryCondition/ConstraintDirichletBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,169 @@
+        (void)local_matrix_size; // unused, but needed for the interface
+        using FemType =
+            NumLib::TemplateIsoparametric<ShapeFunction, ShapeMatricesType>;
+
+        FemType fe(*static_cast<const typename ShapeFunction::MeshElement*>(
+            &_surface_element));
+
+        std::size_t const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        std::vector<
+            typename ShapeMatricesType::ShapeMatrices,
+            Eigen::aligned_allocator<typename ShapeMatricesType::ShapeMatrices>>
+            shape_matrices;
+        shape_matrices.reserve(n_integration_points);
+        _detJ_times_integralMeasure.reserve(n_integration_points);
+        for (std::size_t ip = 0; ip < n_integration_points; ++ip)
+        {
+            shape_matrices.emplace_back(ShapeFunction::DIM, GlobalDim,
+                                        ShapeFunction::NPOINTS);
+            fe.template computeShapeFunctions<NumLib::ShapeMatrixType::N_J>(
+                _integration_method.getWeightedPoint(ip).getCoords(),
+                shape_matrices[ip], GlobalDim, is_axially_symmetric);
+            _detJ_times_integralMeasure.push_back(
+                shape_matrices[ip].detJ * shape_matrices[ip].integralMeasure);
+        }
+    }
+
+    /// Integration for the element with the id \c element_id.
+    /// @param x The global vector containing the values for numerical
+    /// integration.
+    /// @param t The point in time the the integration will be performed.
+    /// @param bulk_mesh The bulk mesh the process is defined on.
+    /// @param getFlux The function of the constraining process used to
+    /// calculate the flux.
+    double integrate(
+        GlobalVector const& x, double const t, MeshLib::Mesh const& bulk_mesh,
+        std::function<Eigen::Vector3d(
+            std::size_t const, MathLib::Point3d const&, double const,
+            GlobalVector const&)> const& getFlux) override
+    {
+        MathLib::Vector3 surface_element_normal;
+        if (_surface_element.getDimension() < 2)
+        {
+            auto const bulk_element_normal =
+                MeshLib::FaceRule::getSurfaceNormal(
+                    bulk_mesh.getElement(_bulk_element_id));
+            MathLib::Vector3 const edge_vector(*_surface_element.getNode(0),
+                                               *_surface_element.getNode(1));
+            surface_element_normal =
+                MathLib::crossProduct(bulk_element_normal, edge_vector);
+        } else {
+            surface_element_normal =
+                MeshLib::FaceRule::getSurfaceNormal(&_surface_element);
+        }
+
+        surface_element_normal.normalize();
+        // At the moment (2018-04-26) the surface normal is not oriented
+        // according to the right hand rule
+        // for correct results it is necessary to multiply the normal with -1
+        surface_element_normal *= -1;
+
+        std::size_t const n_integration_points =
```

Comment:
I'd recommend to use 'auto' here... The return type might change, and is currently an 'unsigned int'.